docs/root/start/sandboxes/single-page-app.rst

```diff
@@ -0,0 +1,648 @@
+   :language: yaml
+   :lines: 34-39
+   :linenos:
+   :lineno-start: 34
+   :emphasize-lines: 3-4
+   :caption: :download:`envoy.yml <_include/single-page-app/envoy.yml>`
+
+On successful authorization/authentication the user is redirected back via this link to the app with the necessary OAuth
+`authorization code <oauth-auth-code_>`__ to proceed:
+
+.. literalinclude:: _include/single-page-app/envoy.yml
+   :language: yaml
+   :lines: 44-50
+   :linenos:
+   :lineno-start: 44
+   :emphasize-lines: 3-5
+   :caption: :download:`envoy.yml <_include/single-page-app/envoy.yml>`
+
+Envoy then uses this authorization code with its client secret to confirm authorization and obtain an access token for the user:
+
+.. literalinclude:: _include/single-page-app/envoy.yml
+   :language: yaml
+   :lines: 50-61
+   :linenos:
+   :lineno-start: 50
+   :emphasize-lines: 3-10
+   :caption: :download:`envoy.yml <_include/single-page-app/envoy.yml>`
+
+.. literalinclude:: _include/single-page-app/secrets/myhub-token-secret.yml
+   :language: yaml
+   :linenos:
+   :emphasize-lines: 6
+   :caption: :download:`myhub-token-secret.yml <_include/single-page-app/secrets/myhub-token-secret.yml>`
+
+Once logged in, you should be able to make queries to the API using the OAuth credentials:
+
+.. image:: /start/sandboxes/_static/spa-resources.png
+   :align: center
+
+.. warning::
+   Envoy's OAuth implementation defaults to triggering the OAuth flow for all paths on the endpoint.
+
+   This can readily trigger an OAuth flood as assets are requested, and doom loops when the OAuth flows fail.
+
+   This can be avoided by restricting the paths that are used by the OAuth flow.
+
+   The sandbox example does this by inverting the
+   :ref:`pass_through_matcher <envoy_v3_api_field_extensions.filters.http.oauth2.v3.OAuth2Config.pass_through_matcher>`
+   to only match on the required OAuth paths.
+
+.. tip::
+   The Myhub OAuth provider does not provide an expiry for issued credentials. Likewise Github may or may
+   not depending on configuration. This is valid in terms of the OAuth2 specification.
+
+   If the authorization provider does not include an expiry, Envoy will, by default, fail the authentication.
+
+   This can be resolved by setting
+   :ref:`default_expires_in <envoy_v3_api_field_extensions.filters.http.oauth2.v3.OAuth2Config.default_expires_in>`:
+
+   .. literalinclude:: _include/single-page-app/envoy.yml
+      :language: yaml
+      :lines: 33-37
+      :linenos:
+      :lineno-start: 33
+      :emphasize-lines: 3
+      :caption: :download:`envoy.yml <_include/single-page-app/envoy.yml>`
+
+.. _install_sandboxes_single_page_app_step_api:
+
+Step 5: Make API queries
+************************
+
+For the sandbox app,
+:ref:`forward_bearer_token <envoy_v3_api_field_extensions.filters.http.oauth2.v3.OAuth2Config.forward_bearer_token>`
+is set, and so Envoy also passes the acquired access token back to the user as a cookie:
```

Comment:
It's my understanding that `forward_bearer_token` governs Envoy to upstream, rather than Envoy to client cookie handling.

Reply:
The cookie is written in any case AFAICT.

Reply:
let me check that again - iirc the browser doesnt get it in that case but i can check again (also wondering what point of cookie would be if its not forwarded)

Reply:
Actually, even more interesting is to follow https://github.com/envoyproxy/envoy/blob/c275fa8d4dfa44fa377b99b4b4eb931f8011c40e/source/extensions/filters/http/oauth2/filter.cc#L636. It sets the OAuth cookie unconditionally, but then an additional bearer cookies for the forward bearer token option. The unconditinoal OAuth cookie though is missing the access token without this option. TIL.

I think what you write is roughly correct then.

Reply:
So I think the cookie exists for the proxy authorization only when not setting `forward_bearer_token`. The cookie is extended with the access token for future accesses when forwarding the token, as Envoy is not remembering it. So, what you have it correct, even though there are cookies in both cases.